### ISSUE TITLE
[dglo on 2016-07-15 10:55:19] : pDAQ: Hubs should send SimplerHits to triggers

### ISSUE THREAD

**Issue by dglo** [dglo/payload#188]
tl;dr : We can shrink the âsimpleâ hit format and reduce the traffic from hubs to trigger handlers by 32-36%

hubs send a condensed 38 byte âsimpleâ hit to their local trigger handler.  The Simple Hit fields are:

Offset/Bytes/Description
0	4	Payload length
4	4	Payload ID
8	8	DAQ time
16	4	trigger type
20	4	configuration ID
24	4	source ID
28	8	DOM ID
36	2	trigger mode

Since its payload ID is 1, Iâm guessing this was probably the first DAQ payload designed.   pDAQ has since introduced the 2-byte channel ID which can encode both the âsource IDâ and âDOM IDâ fields in 2 bytes (as opposed to the 12 bytes currently needed).  Using channel ID would eliminate 10 bytes.

The âconfiguration IDâ is always 0 and is never checked inside the trigger algorithms.

As far as I can tell, the "trigger type" and âtrigger modeâ fields have *always* been identical.  As far as I can tell only the lowest 4 bytes in âtrigger typeâ are ever consulted so we could keep the 2-byte trigger mode field and get rid of the 4-byte trigger type field.

The first 16 bytes are standard and not easily changed unless we rewrite the core payload engines.  If we *could* change this, weâd at most shave off 4-6 bytes by playing tricks to shorten the length and ID fields down to 2-4 bytes but weâd add a small amount of processing time for each payload so Iâm not sure itâs worth it.

So sticking with the low-hanging fruit, we end up with 20 byte payloads which lead to a 48% reduction in network traffic to the trigger handlers.

Offset/Bytes/Description
0	4	Payload length
4	4	Payload ID
8	8	DAQ time
16	2	trigger mode
18    2     channel ID

**Comment by dglo** [comment]
[dglo on 2016-07-22 13:57:01]
svn commit -m"Issue 7873: Unbreak code to find file index number" trigger-testbed/src/main/java/icecube/daq/testbed/SimpleHitFilter.java
Sending        trigger-testbed/src/main/java/icecube/daq/testbed/SimpleHitFilter.java
Transmitting file data .done
Committing transaction...
Committed revision 16186.

svn commit -m"Issue 7873: Separate cached file count from other counts to make the output less confusing" trigger/src/main/java/icecube/daq/trigger/algorithm/AlgorithmStatistics.java
Sending        trigger/src/main/java/icecube/daq/trigger/algorithm/AlgorithmStatistics.java
Transmitting file data .done
Committing transaction...
Committed revision 16187.

svn commit -m"Issue 7873: Add SimplerHit and use by default"
Sending        StringHub/src/main/java/icecube/daq/sender/Sender.java
Sending        StringHub/src/test/java/icecube/daq/sender/SenderTest.java
Sending        daq-request-filler/src/main/java/icecube/daq/reqFiller/StopMarker.java
Sending        eventBuilder-prod/src/test/java/icecube/daq/eventBuilder/test/MockHit.java
Sending        payload/src/main/java/icecube/daq/payload/IHitPayload.java
Sending        payload/src/main/java/icecube/daq/payload/PayloadRegistry.java
Sending        payload/src/main/java/icecube/daq/payload/impl/BaseHitRecord.java
Sending        payload/src/main/java/icecube/daq/payload/impl/DOMHit.java
Sending        payload/src/main/java/icecube/daq/payload/impl/DeltaCompressedHitData.java
Sending        payload/src/main/java/icecube/daq/payload/impl/DeltaHitRecord.java
Sending        payload/src/main/java/icecube/daq/payload/impl/EngineeringHitData.java
Sending        payload/src/main/java/icecube/daq/payload/impl/EventPayload_v5.java
Sending        payload/src/main/java/icecube/daq/payload/impl/PayloadFactory.java
Sending        payload/src/main/java/icecube/daq/payload/impl/SimpleHit.java
Adding         payload/src/main/java/icecube/daq/payload/impl/SimplerHit.java
Sending        payload/src/test/java/icecube/daq/payload/test/MockHit.java
Sending        payload/src/test/java/icecube/daq/payload/test/MockHitData.java
Sending        trigger/src/main/java/icecube/daq/trigger/algorithm/AbstractTrigger.java
Sending        trigger/src/main/java/icecube/daq/trigger/algorithm/CalibrationTrigger.java
Sending        trigger/src/main/java/icecube/daq/trigger/algorithm/ClusterTrigger.java
Sending        trigger/src/main/java/icecube/daq/trigger/algorithm/CylinderTrigger.java
Deleting       trigger/src/main/java/icecube/daq/trigger/algorithm/IniceVolumeTrigger.java
Sending        trigger/src/main/java/icecube/daq/trigger/algorithm/SlowMPTrigger.java
Sending        trigger/src/main/java/icecube/daq/trigger/config/DomSet.java
Sending        trigger/src/main/java/icecube/daq/trigger/config/DomSetFactory.java
Sending        trigger/src/main/java/icecube/daq/trigger/control/HitFilter.java
Deleting       trigger/src/main/java/icecube/daq/trigger/control/StringMap.java
Sending        trigger/src/test/java/icecube/daq/trigger/config/DomSetTest.java
Deleting       trigger/src/test/java/icecube/daq/trigger/control/StringMapTest.java
Sending        trigger/src/test/java/icecube/daq/trigger/control/TriggerManagerTest.java
Sending        trigger/src/test/java/icecube/daq/trigger/test/MockHit.java
Sending        trigger/src/test/resources/config/default-dom-geometry.xml
Transmitting file data .............................done
Committing transaction...
Committed revision 16188.
